Agoda.Frameworks.Grpc/AsyncUnaryCallWrapper.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Threading.Tasks;
+using Grpc.Core;
+using Grpc.Core.Utils;
+
+namespace Agoda.Frameworks.Grpc
+{
+
+    /// <summary>
+    /// Class to flatten Task<AsyncUnaryCall<TResponse>> to plain AsyncUnaryCall<TResponse>
+    /// </summary>
+    class AsyncUnaryCallWrapper<TResponse>
+    {
+
+        private Task<Tuple<TResponse, Metadata, Status, Metadata>> _result;
+
+        public AsyncUnaryCallWrapper(Task<Tuple<TResponse, Metadata, Status, Metadata>> result)
+        {
+            _result = result;
+        }
+
+        public AsyncUnaryCall<TResponse> AsyncUnaryCall
```

Comment:
I prefer a method here.

Reply:
I will change it, but what is the reasoning behind it?

Reply:
https://docs.microsoft.com/en-us/previous-versions/dotnet/netframework-4.0/ms229054(v=vs.100)